CMakeLists.txt

```
@@ -58,7 +58,8 @@ if(${CMAKE_HOST_WIN32})
 	endif()
 endif()
 
-target_link_libraries(rimage PRIVATE crypto)
+find_package(OpenSSL REQUIRED)
```

Nice!

Could you go one step further and try this? It should make openssl 1 fail faster and much more clearly.
```suggestion
find_package(OpenSSL 3...3.9999 REQUIRED)
```

---

This would hopefully provide a much clearer error than the `D:/a/sof/sof/workspace/sof/rimage/src/pkcs1_5.c:93:15: error: implicit declaration of function ‘EVP_PKEY_private_check’; did you mean ‘EVP_PKEY_param_check’?` and similar found in https://github.com/thesofproject/sof/actions/runs/4830399599/jobs/8606585483?pr=7444